contracts/core/SequencerProofVerifier.sol

```diff
@@ -0,0 +1,133 @@
+// SPDX-License-Identifier: Apache-2.0
+/*
+ * Copyright 2024, Polymer Labs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+pragma solidity 0.8.15;
+
+import {RLPReader} from "optimism/libraries/rlp/RLPReader.sol";
+import {RLPWriter} from "optimism/libraries/rlp/RLPWriter.sol";
+import {L1Header, Ics23Proof, OpIcs23Proof} from "../interfaces/IProofVerifier.sol";
+import {ISignatureVerifier} from "../interfaces/ISignatureVerifier.sol";
+import {ECDSA} from "@openzeppelin/contracts/utils/cryptography/ECDSA.sol";
+
+/**
+ * @title OptimisticProofVerifier
+ * @notice Verifies proofs related to Optimistic Rollup state updates
+ * @author Polymer Labs
+ */
+contract SequencerSignatureVerifier is ISignatureVerifier {
+    using RLPReader for RLPReader.RLPItem;
+    using RLPReader for bytes;
+
+    // @notice known L2 Output Oracle contract address to verify state update proofs against
+    address public immutable sequencer;
+
+    error InvalidSequencerSignature();
+
+    constructor(address sequencer_) {
+        sequencer = sequencer_;
+    }
+
+    function verifyStateUpdate(L1Header calldata l1header, bytes32 appHash, uint256 l2Height, bytes calldata signature)
```

Comment:
Can't we provide just the L1 blockhash here instead of the full header and hashing the full header on-chain? I don't think the header fields are used for anything else.

Reply:
technically yes we can and should have this method only take the blockhash. But this is only to save on how much calldata we pass in to this external `verifyStateUpdate` method. 

However, we'd still be computing the blockhash on-chain in the parent `SequencerSoloClient.updateClient`. I think that method needs to take the full header to conform to the `ILightClient` interface?

Reply:
we could fetch it from the l1BlockInfo contract, but that would mean that the peptide sequencer needs to be in sync with the evm chains no?

Reply:
Do we need to check anywhere that this update is consistent with previous state or do we trust each update in isolation because it was signed by our sequencer?

Reply:
I guess it must be this way since that won't necessarily hold if the previous state was based off a different L1 view

Reply:
Would it help to update ILightClient to allow the l1header field to be `bytes` and then allow the LC implementation to cast it to the desired struct?

Reply:
+1 to not using the full L1 header here. We had that field in the original "merkle proof client" because it was the only way to link the L1 state root to a L1 hash that we could then compare with the chain's view of L1 (via L1Block contract) but in here we trust the L1 hash as long as the signature matches.

Reply:
would something like[ this commit](https://github.com/open-ibc/vibc-core-smart-contracts/pull/198/commits/62a1e2645af3d7100ed64edf4ba0decdd31217ed) work?

Reply:
should be added now that we have rebased the generic bytes arg